src/specmap/helpers.js

```diff
@@ -58,7 +58,7 @@ export function generateAbsoluteRefPatches(
   const patches = [];
 
   traverse(obj).forEach(function callback() {
-    if (targetKeys.indexOf(this.key) > -1) {
+    if (targetKeys.indexOf(this.key) > -1 && typeof this.node === 'string') {
```

Comment:
Per the doc, `$ref: 1`, where `1` is a number, should cause an error during the resolving.
>If the string value does not conform URI syntax rules, it causes an error during the resolving. 

Ok, and will do, on the refactoring suggestions.

Reply:
> Per the doc, $ref: 1, where 1 is a number, should cause an error during the resolving.

Well that's the exact thing I'm trying to pointing out. It will not cause an error because it will not be considered a JSON Pointer after our code-change (expecting only strings). If you'll create a test, you'll see. I'm not sure how to solve this except checking for logical disjunction of `isNumber` with `isString

Reply:
This PR is not trying to fix the resolver with regards to JSON Pointer values. This PR stops the resolver from continuing to traverse and "absolutify" references when `$ref` is not actually a string. Can you create a new ticket for the corner case you described?

Reply:
Then I'll be raising an issue for code-change that doesn't exist on master yet ;] I've raised my concern here, I'll leave it up to you to decide what to do with it. The solution might be as simple as to recommend user to wrap YAML properties into double quotes if they expect to be parted as string (but I guess that's already implied by YAML format itself). So this may be a non issue after-all.